alien-module/control.lua

```diff
@@ -73,45 +73,52 @@ function update_gui()
     end
 end
 
-function update_modules_in_module_slot(entities, level)
-    for _, entity in ipairs(entities) do
-        local moduleInventory = entity.get_module_inventory()
+function update_modules(entities, entityType)
+    for _, entity in pairs(entities) do
+        local inventory --what type of inventory does this entity have?
+
+        if entityType == "chest" then
+            inventory = entity.get_inventory(defines.inventory.chest) --grab a chest's inventory
+        elseif entityType == "machine" then
+            inventory = entity.get_module_inventory() --grab a machine's inventory
+        elseif entityType == "player" then
+            inventory = entity.get_inventory(defines.inventory.player_main) --grab a player's inventory
+        else
+            return --error entity type not defined
+        end
 
-        for i = 1, 10, 1 do
+        for i = 1, #inventory, 1 do --loop through all of the entity's inventory slots
```

Comment:
what is #inventory?

Reply:
oh i see its the length of the array

Reply:
it's the number (#) of slots in the variable [inventory]

Reply:
https://lua-api.factorio.com/latest/LuaInventory.html

it shows it here

operator # :: uint [R] - Get the number of slots in this inventory.

Reply:
where do you get the lua formatter?

Reply:
i use the intellij standard lua formatter